### PR TITLE
TST: sparse.linalg.lobpcg: add unit tests for accuracy

### LINKED ISSUE #19355: ENH: add unit tests for accuracy matching benchmark tests for performance
### Is your feature request related to a problem? Please describe.

https://github.com/scipy/scipy/pull/18954 has introduced a few new benchmark tests for iterative eigensolvers where their convergence is not necessarily guaranteed. Corresponding unit tests should be thus added to the check convergence. Similarly for pre-existing benchmark tests treated in https://github.com/scipy/scipy/pull/19292. 

### Describe the solution you'd like.

New unit tests matching the existing benchmark tests but checking for convergence.

### Describe alternatives you've considered.

https://github.com/scipy/scipy/pull/18954 includes accuracy asserts to be on a safe side, but performance benchmarks is not a good place to also assess the accuracy.   

### Additional context (e.g. screenshots, GIFs)

_No response_

### PR DESCRIPTION
**Pull request by lobpcg**
#### Reference issue
Closes gh-19355

#### What does this implement/fix?
See gh-19355

### CODE REVIEW

**Comment by lobpcg** [comment]
@tylerjereddy @rkern - ready for review

File: benchmarks/benchmarks/sparse_linalg_lobpcg.py [hunk 1]
```diff
@@ -83,10 +75,8 @@ def a(x):
             c = cholesky_banded(self.Ab.astype(np.float32))
             with warnings.catch_warnings():
                 warnings.simplefilter("ignore")
-                el, _ = lobpcg(self.Ac, X, self.Bc, M=a, tol=1e-4,
+                _, _ = lobpcg(self.Ac, X, self.Bc, M=a, tol=1e-4,
```
Comment (j-bowhay) [inline benchmarks/benchmarks/sparse_linalg_lobpcg.py:78]: here and elsewhere
```suggestion
                lobpcg(self.Ac, X, self.Bc, M=a, tol=1e-4,
```

File: scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py [hunk 2]
```diff
@@ -52,53 +57,24 @@ def ElasticRod(n):
...
-        compare_solutions(A, B, 10)
-
-
-def test_ElasticRod():
-    A, B = ElasticRod(20)
-    with pytest.warns(UserWarning, match="Exited at iteration"):
-        compare_solutions(A, B, 2)
-
-
-def test_MikotaPair():
```
Comment (j-bowhay) [inline scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py:97]: Other than switching to the new MikotaPair why are the changes here needed?

**Review by j-bowhay — CHANGES_REQUESTED** [review]

File: scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py (hunk 2 above)
Comment (lobpcg) [reply scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py:97]: I forgot already. I guess mostly just general cleaning and unification/beatification.

**Comment by lobpcg** [comment]
Linux Meson tests / Meson build (3.12-dev, false) (pull_request) Failing is unrelated to this PR.

**Comment by lobpcg** [comment]
Waiting for reviews...

**Comment by lobpcg** [comment]
This PR is completed 2 months ago now. Could someone review it and consider merging?

**Comment by lobpcg** [comment]
It is so old that has become obsolete and can't be merged without extra efforts any more... Is there interest to finish this? I already forgot about it.

**Comment by lobpcg** [comment]
moved to two separate PRs
https://github.com/scipy/scipy/pull/21220
https://github.com/scipy/scipy/pull/21221/files

**Pull request closed without merging**